libcxx/include/support/atomic/atomic_cuda.h

```diff
@@ -376,6 +376,34 @@ __host__ __device__
     )
 }
 
+template <typename _Tp, typename _Delta, int _Sco, bool _Ref>
+__host__ __device__
+ _Tp __cxx_atomic_fetch_max(__cxx_atomic_base_heterogeneous_impl<_Tp, _Sco, _Ref> volatile* __a, _Delta __val, memory_order __order) {
+    _Tp __expected = __cxx_atomic_load(__a, memory_order_relaxed);
```

Comment:
This needs to be scoped, no?

Reply:
`_Sco` is carried by `__a` for the `__cxx_atomic_` layer.

Reply:
Ah! that makes a lot of sense.